diff --git a/node_modules/expo-share-intent/android/src/main/java/expo/modules/shareintent/ExpoShareIntentModule.kt b/node_modules/expo-share-intent/android/src/main/java/expo/modules/shareintent/ExpoShareIntentModule.kt
--- a/node_modules/expo-share-intent/android/src/main/java/expo/modules/shareintent/ExpoShareIntentModule.kt
+++ b/node_modules/expo-share-intent/android/src/main/java/expo/modules/shareintent/ExpoShareIntentModule.kt
@@ -153,6 +153,13 @@ class ExpoShareIntentModule : Module() {
                     } else {
                         notifyError("empty uris array for file sharing: " + intent.action)
                     }
+                } else if (intent.action == Intent.ACTION_VIEW) {
+                    val uri = intent.data
+                    if (uri != null) {
+                        notifyShareIntent(mapOf( "files" to arrayOf(getFileInfo(uri)), "type" to "file"))
+                    } else {
+                        notifyError("empty uri for file opening: " + intent.action)
+                    }
                 } else {
                     notifyError("Invalid action for file sharing: " + intent.action)
                 }
